src/core/quantum_gate.jl

```diff
@@ -666,6 +833,24 @@ get_operator(gate::Pi8) = pi_8(gate.type)
 
 get_inverse(gate::Pi8) = pi_8_dagger(gate.target[1],gate.type)
 
+"""
+    pi_8_diag(target)
+
+Return a π/8 `Gate` (also known as a ``T`` `Gate`), which applies the [`pi_8_diag()`](@ref) `DiagonalOperator` to the `target` qubit.
+"""
+pi_8_diag(target::Integer) = Pi8_Diag([target])
+
+struct Pi8_Diag <: AbstractGate
+    target_list::Vector{<:Integer}
```

Comment:
1. There is only one target so lets not have the `Vector`.
2. This can simply be an `Int` unless there is a benefit to having it templated?